### PR TITLE
PFW-1060: added new function belt test under calibration menu

### PR DESCRIPTION
**Pull request by arons**
it takes quite some times to check belt status passing trough self test.
That is why I've added a specific function for that.

### CODE REVIEW

**Comment by arons** [comment]
If you want I can also move the functionality to other menu places.
For example directly under Support->Belt status.
Let me know your wishes.

**Comment by F0x06** [comment]
Good job, this is awesome ! Works perfectly, this really needs to be merged by Prusa.

I implemented that myself some weeks ago, you did a better job than me with real time display etc, thanks :)

**Comment by DRracer** [comment]
@arons good job, we'll merge this into 3.9

File: Firmware/ultralcd.cpp [hunk 1]
```diff
@@ -6941,6 +6948,63 @@ void lcd_sdcard_menu()
...
+{
+    lcd_belttest();
+    menu_back_if_clicked();
+}
+void lcd_belttest_print(const char* msg, uint16_t X, uint16_t Y)
+{
+    lcd_clear();
+    lcd_printf_P(
+              _N(
+                 ESC_H(0,0)
```
Comment (leptun) [inline Firmware/ultralcd.cpp]: @DRracer If you merge this and also the lcd improvements changes you need to get rid of that escape code. It is not needed in this case as lcd_clear() will also set the cursor to 0,0. Just remove line 6962.

**Review by DRracer — APPROVED** [review]

**Comment by vintagepc** [comment]
This feature does not work on my printer. If I try to use it, Z moves up ~20mm, and I get a "Selftest error" on screen instructing me to check X axis length. X never attempts to move, and the updated status value is 0. I filed a bug report #2356

**Pull request merged**